src/tallies/tally_scoring.cpp

```diff
@@ -0,0 +1,2347 @@
+    case N_4N:
+    case N_GAMMA:
+    case N_P:
+    case N_A:
+      if (tally.estimator_ == ESTIMATOR_ANALOG) {
+        // Check if the event MT matches
+        if (p->event_mt_ != score_bin) continue;
+        score = p->wgt_last_ * flux;
+      } else {
+        int m;
+        switch (score_bin) {
+        case N_GAMMA: m = 0; break;
+        case N_P: m = 1; break;
+        case N_A: m = 2; break;
+        case N_2N: m = 3; break;
+        case N_3N: m = 4; break;
+        case N_4N: m = 5; break;
+        }
+        if (i_nuclide >= 0) {
+          score = p->neutron_xs_[i_nuclide].reaction[m] * atom_density
+            * flux;
+        } else {
+          score = 0.;
+          if (p->material_ != MATERIAL_VOID) {
+            const Material& material {*model::materials[p->material_]};
+            for (auto i = 0; i < material.nuclide_.size(); ++i) {
+              auto j_nuclide = material.nuclide_[i];
+              auto atom_density = material.atom_density_(i);
+              score += p->neutron_xs_[j_nuclide].reaction[m]
+                * atom_density * flux;
+            }
+          }
+        }
+      }
+      break;
+
+
+    case HEATING:
+      score = 0.;
+      if (p->type_ == Particle::Type::neutron) {
+        score = score_neutron_heating(p, tally, flux, HEATING,
+            i_nuclide, atom_density);
+      } else if (p->type_ == Particle::Type::photon) {
+        if (tally.estimator_ == ESTIMATOR_ANALOG) {
+          // Score direct energy deposition in the collision
+          score = E - p->E_;
+          // We need to substract the energy of the secondary particles since
+          // they will be transported individually later
+          for (auto i = 0; i < p->n_bank_second_; ++i) {
+            auto i_bank = simulation::secondary_bank.size() - p->n_bank_second_ + i;
+            const auto& bank = simulation::secondary_bank[i_bank];
+            if (bank.particle == Particle::Type::photon ||
+                bank.particle == Particle::Type::neutron) {
+              score -= bank.E;
+            } else if (bank.particle == Particle::Type::positron) {
+              // Annihilation of the positron will produce two new photons
+              score -= 2*MASS_ELECTRON_EV;
+            }
+          }
+          score *= p->wgt_last_ * flux;
+        } else {
+          // Calculate photon heating cross section on-the-fly
+          if (i_nuclide >= 0) {
+            // Find the element corresponding to the nuclide
+            auto name = data::nuclides[i_nuclide]->name_;
+            std::string element = to_element(name);
+            int i_element = data::element_map[element];
+            auto& heating {data::elements[i_element].heating_};
+            auto i_grid = p->photon_xs_[i_element].index_grid;
+            auto f = p->photon_xs_[i_element].interp_factor;
+            score = std::exp(heating(i_grid) + f * (heating(i_grid+1) -
+              heating(i_grid))) * atom_density * flux;
+          } else {
```

Comment:
Optional: You can collapse this `else` and the `if` below into a `else if` statement.  E.g.:

```
} else if(p-material_ != MATERIAL_VOID) {
    ...
}
```